modules/imgproc/src/distransform.cpp

```diff
@@ -796,6 +796,17 @@ void cv::distanceTransform( InputArray _src, OutputArray _dst, OutputArray _labe
                     ippFree( pBuffer );
                     if (status>=0)
                     {
+                        static const float correctionDiff = 1.0f / (1 << 12);
+                        for (int i = 0; i < dst.rows; ++i)
+                        {
+                            float* row = dst.ptr<float>(i);
+                            for (int j = 0; j < dst.cols; ++j)
+                            {
+                                float rounded = cvRound(row[j]);
+                                if (fabs(row[j] - rounded) <= correctionDiff)
+                                    row[j] = rounded;
+                            }
+                        }
```

Comment:
I recommend you to add comment with link to the original bug and couple of words why the loop is needed. The reason of "clipping" is not obvious.

Reply:
Sure, added the comment